src/EditorFeatures/CSharpTest/Formatting/FormattingEngineTests.cs

```diff
@@ -2289,9 +2289,129 @@ class Test
             Assert.Single(annotatedTrivia);
         }
 
-        private static void AssertFormatAfterTypeChar(string code, string expected, Dictionary<OptionKey2, object> changedOptionSet = null)
+        [WpfFact]
+        public void FormatUserDefinedOperator()
         {
-            using var workspace = TestWorkspace.CreateCSharp(code);
+            var code = @"$$
+class C
+{
+    public static C operator + ( C x, C y){
+    }
+}";
+
+            var expected = @"$$
+class C
+{
+    public static C operator +(C x, C y)
+    {
+    }
+}";
+
+            AssertFormatWithView(expected, code);
+        }
+
+        [WpfFact]
+        public void FormatUserDefinedExplicitCastOperator()
+        {
+            var code = @"$$
+class C
+{
+    public static explicit operator C ( int x){
+    }
+}";
+
+            var expected = @"$$
+class C
+{
+    public static explicit operator C(int x)
+    {
+    }
+}";
+
+            AssertFormatWithView(expected, code);
+        }
+
+        [WpfFact]
+        public void FormatUserDefinedOperatorOnType()
+        {
+            var code = @"
+interface I1
+{
+    abstract static I1 operator + ( I1 x, I1 y);$$
+}";
+
+            var expected = @"
+interface I1
+{
+    abstract static I1 operator +(I1 x, I1 y);
+}";
+
+            AssertFormatAfterTypeChar(code, expected);
+        }
+
+        [WpfFact]
+        public void FormatUserDefinedCheckedOperator()
+        {
+            var code = @"$$
+class C
+{
+    public static C operator checked + ( C x, C y){
+    }
+}";
+
+            var expected = @"$$
+class C
+{
+    public static C operator checked +(C x, C y)
+    {
+    }
+}";
+
+            AssertFormatWithView(expected, code, parseOptions: CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.Preview));
+        }
+
+        [WpfFact]
+        public void FormatUserDefinedExplicitCheckedCastOperator()
+        {
+            var code = @"$$
+class C
+{
+    public static explicit operator checked C ( int x){
+    }
+}";
+
+            var expected = @"$$
+class C
+{
+    public static explicit operator checked C(int x)
+    {
+    }
+}";
+
+            AssertFormatWithView(expected, code, parseOptions: CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.Preview));
+        }
+
+        [WpfFact]
+        public void FormatUserDefinedCheckedOperatorOnType()
+        {
+            var code = @"
+interface I1
+{
+    abstract static I1 operator checked + ( I1 x, I1 y);$$
```

Comment:
>abstract static I1 operator checked + ( I1 x, I1 y); [](http://example.com/codeflow?start=4&length=52)

Consider also testing unary and explicit conversion operators for completeness.

Reply:
good call!  Found a bug with format on type for cast operators, see https://github.com/dotnet/roslyn/pull/60375/commits/6e3a5930aab38b22f8e1d5fe7e9a11cfdcaecfd2#diff-8dbfd863823b61ac2d84d6331f2b45d71ae69ea52d48b098ffb08b922abc7ce7R129